docs/reactive-foundations.qmd

```diff
@@ -0,0 +1,151 @@
+---
+title: "Foundations"
+format: html
+aliases:
+  - reactive-programming.html
+  - reactive-calculations.html
+  - reactive-events.html
+  - reactive-values.html
+---
+
+At the heart of Shiny is a reactive programming framework.
+We often refer to this framework as "good magic" because it's easy to get started with, but also decomposes into simple pieces which combine in powerful ways.
+
+In [Quick Start](quick-start.qmd), we covered the most common form of reactivity: changes in `input` causing relevant `render` functions to re-execute (aka **invalidate**).
+More generally, Shiny knows to re-execute **reactive functions** (e.g., `render` functions) when relevant **reactive dependencies** (e.g., `input`) change.
```

Comment:
That's sort of the goal of this article: to loosely define and provide more examples of reactive functions and dependencies. I'd be happy to consider adding a more formal definition if you can think of one, but it's probably too late to avoid the terms altogether